CONTRIBUTING.rst

```diff
@@ -0,0 +1,264 @@
+    setvirtualenvproject
+
+Install Python requirements::
+
+    pip install pipenv
+    pipenv install --dev
+
+Copy the ``.env`` file somewhere that will be sourced when you need it::
+
+    cp env.local.example $VIRTUAL_ENV/bin/postactivate
+
+Edit this file to change ``DJANGO_SECRET_KEY`` and ``DJANGO_HASHID_SALT`` to any
+two different arbitrary string values. Also set ``DB_ENCRYPTION_KEY``::
+
+    python
+    from cryptography.fernet import Fernet
+    Fernet.generate_key()
+
+This will output a bytestring, e.g. ``b'mystring='``. Copy only the contents of
+``'...'``, e.g. ``export DB_ENCRYPTION_KEY="mystring="``.
+
+Finally, add the following environment variables::
+
+    export BUCKETEER_AWS_ACCESS_KEY_ID=...
+    export BUCKETEER_AWS_SECRET_ACCESS_KEY=...
+    export BUCKETEER_BUCKET_NAME=...
+
+Now run ``workon metashare`` again to set those environment variables.
+
+Your ``PATH`` (and environment variables) will be updated when you ``workon
+metashare`` and restored when you ``deactivate``. This will make sure that
+whenever you are working on the project, you use the project-specific version of
+Node instead of any system-wide Node you may have.
+
+**All of the remaining steps assume that you have the virtualenv activated.**
+(``workon metashare``)
+
+.. _virtualenvwrapper: https://virtualenvwrapper.readthedocs.io/en/latest/
+
+Installing JavaScript requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The project-local version of `Node.js`_ can be downloaded and unpacked locally
+(in the git-ignored ``node/`` directory), so you don't have to install it
+system-wide (and possibly conflict with other projects wanting other Node
+versions).
+
+To download and install the project-local version of Node (and `yarn`_)::
+
+    bin/unpack-node
+
+If you can run ``which node`` and see a path inside your project directory
+ending with ``.../node/bin/node``, then you've got it set up right and can move
+on.
+
+Then use ``yarn`` to install dependencies::
+
+    yarn
+
+.. _Node.js: http://nodejs.org
+.. _yarn: https://yarnpkg.com/
+
+Setting up the database
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Assuming you have `Postgres <https://www.postgresql.org/download/>`_ installed
+and running locally::
+
+    createdb metashare
+
+Then run the initial migrations::
+
+    python manage.py migrate
+
+Running the server
+~~~~~~~~~~~~~~~~~~
+
+The local development server requires `Redis <https://redis.io/>`_ to manage
+background worker tasks. If you can successfully run ``redis-cli ping`` and see
+output ``PONG``, then you have Redis installed and running. Otherwise, run
+``brew install redis`` (followed by ``brew services start redis``) or refer to
+the `Redis Quick Start`_.
+
+To run the local development server::
+
+    yarn serve
+
+The running server will be available at `<http://localhost:8080/>`_.
+
+.. _Redis Quick Start: https://redis.io/topics/quickstart
+
+Logging in with Salesforce
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think this should probably be copied above to the Docker setup too, with commands edited to work through Docker.